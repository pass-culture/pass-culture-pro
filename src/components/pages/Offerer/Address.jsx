import React from 'react'
import { Field } from 'react-final-form'

const required = value => {
  return value ? undefined : 'Ce champs est requis'
}

const Address = () => (
  <Field
    name="address"
    validate={required}
  >
    {({ input }) => {
      return (
        <div className='field text-field is-label-aligned'>
          <label
            className='field-label'
            htmlFor="offerer__address"
          >
            <span>
              {'Si√®ge social : '}
            </span>
          </label>
          <div className="field-control">
            <div className="field-value flex-columns items-center">
              <div className='field-inner flex-columns items-center'>
                <input
                  className="input is-normal"
                  id="offerer__address"
                  readOnly
                  type="text"
                  {...input}
                />
              </div>
            </div>
          </div>
        </div>
      )
    }}
  </Field>
)

export default Address
