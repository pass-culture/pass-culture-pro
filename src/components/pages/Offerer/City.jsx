import React from 'react'
import { Field } from 'react-final-form'

const City = () => (
  <Field
    name="city"
  >
    {({ input }) => {
      return (
        <div className='field text-field is-label-aligned'>
          <label
            className='field-label'
            htmlFor="city"
          >
            <span>
              {'Ville : '}
            </span>
          </label>
          <div className="field-control">
            <div className="field-value flex-columns items-center">
              <div className='field-inner flex-columns items-center'>
                <input
                  className="input is-normal"
                  id="city"
                  readOnly
                  type="text"
                  {...input}
                />
              </div>
            </div>
          </div>
        </div>
      )
    }}
  </Field>
)

export default City
