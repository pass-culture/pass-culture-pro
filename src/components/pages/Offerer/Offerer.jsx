import classnames from 'classnames'
import get from 'lodash.get'
import { Field, Form, showNotification } from 'pass-culture-shared'
import React, { Component } from 'react'
import { requestData } from 'redux-saga-data'

import CreationControl from './CreationControl/CreationControl'
import ModificationControl from './ModificationControl/ModificationControl'
import HeroSection from '../../layout/HeroSection/HeroSection'
import Main from '../../layout/Main'
import { offererNormalizer } from '../../../utils/normalizers'
import { formatPatch } from '../../../utils/formatPatch'

const OFFERER_CREATION_PATCH_KEYS = ['address', 'city', 'name', 'siren', 'postalCode']
const OFFERER_MODIFICATION_PATCH_KEYS = ['bic', 'iban', 'rib']

class Offerer extends Component {
  onHandleDataRequest = (handleSuccess, handleFail) => {
    const {
      dispatch,
      match: {
        params: { offererId },
      },
      query,
    } = this.props
    const { isCreatedEntity } = query.context()

    if (!isCreatedEntity) {
      dispatch(
        requestData({
          apiPath: `/offerers/${offererId}`,
          handleSuccess,
          handleFail,
          normalizer: offererNormalizer,
        })
      )

      dispatch(requestData({ apiPath: `/userOfferers/${offererId}` }))

      return
    }

    handleSuccess()
  }

  onHandleFail = () => {
    const { dispatch } = this.props
    dispatch(
      showNotification({
        text: 'Vous étes déjà rattaché à cette structure.',
        type: 'danger',
      })
    )
  }

  onHandleSuccess = () => {
    const { dispatch, history } = this.props
    const { isCreatedEntity } = this.props

    history.push('/structures')

    const text = isCreatedEntity
      ? 'Votre structure a bien été enregistrée, elle est en cours de validation.'
      : 'Les modifications sur votre structure ont bien été prises en compte.'

    dispatch(
      showNotification({
        text,
        type: 'success',
      })
    )
  }

  handleDisabling = offererName => () => !offererName

  formatPatch = patchConfig => patch =>
    formatPatch(patch, patchConfig, OFFERER_CREATION_PATCH_KEYS, OFFERER_MODIFICATION_PATCH_KEYS)

  render() {
    const { adminUserOfferer, offerer, query, offererName } = this.props
    const { isCreatedEntity, isModifiedEntity, readOnly } = query.context()
    const areSirenFieldsVisible = get(offerer, 'id') || offererName
    const areBankInfosReadOnly = readOnly || !adminUserOfferer

    const patchConfig = { isCreatedEntity, isModifiedEntity }

    return (
      <Main
        backTo={{ label: 'Vos structures juridiques', path: '/structures' }}
        handleDataRequest={this.onHandleDataRequest}
        name="offerer"
      >
        <HeroSection
          subtitle={get(offerer, 'name')}
          title="Structure"
        >
          <p className="subtitle">
            {'Détails de la structure rattachée, des lieux et des fournisseurs de ses offres.'}
          </p>
        </HeroSection>

        <Form
          action={`/offerers/${get(offerer, 'id') || ''}`}
          className="section"
          formatPatch={this.formatPatch(patchConfig)}
          handleFail={this.onHandleFail}
          handleSuccess={this.onHandleSuccess}
          name="offerer"
          patch={offerer}
          readOnly={readOnly}
        >
          <div className="section">
            <div className="field-group">
              <Field
                disabling={this.handleDisabling(offererName)}
                label="SIREN"
                name="siren"
                readOnly={get(offerer, 'id')}
                required
                type="siren"
              />
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="Désignation"
                name="name"
                readOnly
                required
              />
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="Siège social"
                name="address"
                readOnly
              />
            </div>
          </div>
          <div className="section">
            <h2 className="main-list-title">
              {'INFORMATIONS BANCAIRES'}
              <span className="is-pulled-right is-size-7 has-text-grey">
                {readOnly &&
                  !adminUserOfferer &&
                  "Vous avez besoin d'être administrateur de la structure pour modifier ces informations."}
              </span>
            </h2>
            <div className="field-group">
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="BIC"
                name="bic"
                readOnly={areBankInfosReadOnly}
                type="bic"
              />
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="IBAN"
                name="iban"
                readOnly={areBankInfosReadOnly}
                type="iban"
              />
            </div>
            {isCreatedEntity ? <CreationControl /> : <ModificationControl {...this.props} />}
          </div>
        </Form>
      </Main>
    )
  }
}

export default Offerer
