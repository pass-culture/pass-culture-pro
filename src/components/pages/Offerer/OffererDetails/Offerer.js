export class Offerer {
  constructor(offerer = {}, adminUserOfferer) {
    this.id = offerer.id || undefined // pass-culture-shared awaits undefined to condition the submit method (PATCH vs POST)
    this.siren = offerer.siren || ''
    this.name = offerer.name || ''
    this.address = offerer.address || ''
    this.bic = offerer.bic || ''
    this.iban = offerer.iban || ''
    this.adminUserOfferer = adminUserOfferer
  }

  get areBankInformationProvided () {
    return !!(this.bic && this.iban)
  }
}
