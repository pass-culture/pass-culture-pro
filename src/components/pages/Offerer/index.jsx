import classnames from 'classnames'
import get from 'lodash.get'
import {
  CancelButton,
  Field,
  Form,
  requestData,
  showNotification,
  SubmitButton,
  withLogin,
} from 'pass-culture-shared'
import React, { Component, Fragment } from 'react'
import { connect } from 'react-redux'
import { NavLink, withRouter } from 'react-router-dom'
import { compose } from 'redux'

import HeroSection from '../../layout/HeroSection'
import VenueItem from './VenueItem'
import Main from '../../layout/Main'
import offererSelector from '../../../selectors/offerer'
import selectUserOffererByOffererIdAndUserIdAndRightsType from '../../../selectors/selectUserOffererByOffererIdAndUserIdAndRightsType'
import selectPhysicalVenuesByOffererId from '../../../selectors/selectPhysicalVenuesByOffererId'
import { offererNormalizer } from '../../../utils/normalizers'

import {
  formatPatch,
  OFFERER_NEW_PATCH_KEYS,
  OFFERER_EDIT_PATCH_KEYS,
} from '../../../utils/formatPatch'

class OffererPage extends Component {
  constructor() {
    super()
    this.state = {
      isNew: false,
      isReadOnly: true,
    }
  }

  static getDerivedStateFromProps(nextProps) {
    const {
      location: { search },
      match: {
        params: { offererId },
      },
    } = nextProps

    const isEdit = search.includes('modifie')
    const isNew = offererId === 'nouveau'
    const isReadOnly = !isNew && !isEdit
    return {
      isEdit,
      isNew,
      isReadOnly,
    }
  }

  handleDataRequest = (handleSuccess, handleFail) => {
    const {
      dispatch,
      match: {
        params: { offererId },
      },
    } = this.props

    if (offererId !== 'nouveau') {
      dispatch(
        requestData('GET', `offerers/${offererId}`, {
          handleSuccess,
          handleFail,
          normalizer: offererNormalizer,
        })
      )

      dispatch(requestData('GET', `userOfferers/${offererId}`))

      return
    }

    handleSuccess()
  }

  handleSuccess = () => {
    const { dispatch, history } = this.props
    const { isNew } = this.props

    history.push('/structures')

    const text = isNew
      ? 'Votre structure a bien été enregistrée, elle est en cours de validation.'
      : 'Les modifications sur votre structure ont bien été pris en compte'

    dispatch(
      showNotification({
        text,
        type: 'success',
      })
    )
  }

  onAddProviderClick = () => {
    this.setState({ isNewProvider: true })
  }

  render() {
    const { adminUserOfferer, offerer, sirenName, venues } = this.props
    const { isEdit, isNew, isReadOnly } = this.state
    const { id } = offerer || {}
    const areSirenFieldsVisible = get(offerer, 'id') || sirenName
    const areBankInfosReadOnly = isReadOnly || !adminUserOfferer

    const patchConfig = { isNew, isEdit }

    const $newControl = (
      <div
        className="field is-grouped is-grouped-centered"
        style={{ justifyContent: 'space-between' }}>
        <div className="control">
          <NavLink className="button is-secondary is-medium" to="/structures">
            Retour
          </NavLink>
        </div>
        <div className="control">
          <SubmitButton className="button is-primary is-medium">
            Valider
          </SubmitButton>
        </div>
      </div>
    )

    const $editControl = (
      <Fragment>
        <div className="control">
          {isReadOnly ? (
            adminUserOfferer && (
              <NavLink
                className="button is-secondary is-medium"
                to={`/structures/${id}?modifie`}>
                Modifier les informations
              </NavLink>
            )
          ) : (
            <div
              className="field is-grouped is-grouped-centered"
              style={{ justifyContent: 'space-between' }}>
              <div className="control">
                <CancelButton
                  className="button is-secondary is-medium"
                  to={`/structures/${id}`}>
                  Annuler
                </CancelButton>
              </div>
              <div className="control">
                <SubmitButton className="button is-primary is-medium">
                  Valider
                </SubmitButton>
              </div>
            </div>
          )}
        </div>
        <br />
        <div className="section">
          <h2 className="main-list-title">LIEUX</h2>
          <ul className="main-list venues-list">
            {venues.map(v => (
              <VenueItem key={v.id} venue={v} />
            ))}
          </ul>
          <div className="has-text-centered">
            <NavLink
              to={`/structures/${get(offerer, 'id')}/lieux/nouveau`}
              className="button is-secondary is-outlined">
              + Ajouter un lieu
            </NavLink>
          </div>
        </div>
      </Fragment>
    )

    return (
      <Main
        backTo={{ label: 'Vos structures juridiques', path: '/structures' }}
        name="offerer"
        handleDataRequest={this.handleDataRequest}>
        <HeroSection subtitle={get(offerer, 'name')} title="Structure">
          <p className="subtitle">
            Détails de la structure rattachée, des lieux et des fournisseurs de
            ses offres.
          </p>
        </HeroSection>

        <Form
          action={`/offerers/${get(offerer, 'id') || ''}`}
          name="offerer"
          className="section"
          formatPatch={patch =>
            formatPatch(
              patch,
              patchConfig,
              OFFERER_NEW_PATCH_KEYS,
              OFFERER_EDIT_PATCH_KEYS
            )
          }
          handleSuccess={this.handleSuccess}
          patch={offerer}
          readOnly={isReadOnly}>
          <div className="section">
            <div className="field-group">
              <Field
                disabling={() => !sirenName}
                label="SIREN"
                name="siren"
                readOnly={get(offerer, 'id')}
                required
                type="siren"
              />
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="Désignation"
                name="name"
                readOnly
                required
              />
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="Siège social"
                name="address"
                readOnly
              />
            </div>
          </div>
          <div className="section">
            <h2 className="main-list-title">
              INFORMATIONS BANCAIRES
              <span className="is-pulled-right is-size-7 has-text-grey">
                {isReadOnly &&
                  !adminUserOfferer &&
                  "Vous avez besoin d'être administrateur de la structure pour editer ces informations."}
              </span>
            </h2>
            <div className="field-group">
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="BIC"
                name="bic"
                type="bic"
                readOnly={areBankInfosReadOnly}
              />
              <Field
                className={classnames({
                  'is-invisible': !areSirenFieldsVisible,
                })}
                isExpanded
                label="IBAN"
                name="iban"
                type="iban"
                readOnly={areBankInfosReadOnly}
              />
            </div>
            {isNew ? $newControl : $editControl}
          </div>
        </Form>
      </Main>
    )
  }
}

function mapStateToProps(state, ownProps) {
  const offererId = ownProps.match.params.offererId
  const { id: userId } = state.user || {}
  return {
    adminUserOfferer: selectUserOffererByOffererIdAndUserIdAndRightsType(
      state,
      offererId,
      userId,
      'admin'
    ),
    offerer: offererSelector(state, offererId),
    sirenName: get(state, 'form.offerer.name'),
    user: state.user,
    venues: selectPhysicalVenuesByOffererId(state, offererId),
  }
}

export default compose(
  withLogin({ failRedirect: '/connexion' }),
  withRouter,
  connect(mapStateToProps)
)(OffererPage)
