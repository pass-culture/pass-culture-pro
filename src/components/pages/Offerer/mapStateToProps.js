import get from 'lodash.get'

import selectOffererById from 'selectors/selectOffererById'
import selectUserOffererByOffererIdAndUserIdAndRightsType from 'selectors/selectUserOffererByOffererIdAndUserIdAndRightsType'
import selectPhysicalVenuesByOffererId from 'selectors/selectPhysicalVenuesByOffererId'

const mapStateToProps = (state, ownProps) => {
  const { currentUser, match } = ownProps
  const {
    params: { offererId },
  } = match
  const { id: currentUserId } = currentUser || {}

  return {
    adminUserOfferer: selectUserOffererByOffererIdAndUserIdAndRightsType(
      state,
      offererId,
      currentUserId,
      'admin'
    ),
    offerer: selectOffererById(state, offererId),
    offererName: get(state, 'form.offerer.name'),
    venues: selectPhysicalVenuesByOffererId(state, offererId),
  }
}

export default mapStateToProps
