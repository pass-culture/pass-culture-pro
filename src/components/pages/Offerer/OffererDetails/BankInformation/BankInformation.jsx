import { Offerer } from '../Offerer'
import PropTypes from 'prop-types'
import { DEMARCHES_SIMPLIFIEES_OFFERER_RIB_UPLOAD_PROCEDURE_URL } from '../../../../../utils/config'
import React, { Fragment } from 'react'
import Icon from '../../../../layout/Icon'

const BankInformation = ({ offerer }) => (
  <div className="section op-content-section bank-information">
    <h2 className="main-list-title">
      {'Coordonnées bancaires de la structure'}
    </h2>

    {offerer.areBankInformationProvided ? (
      <Fragment>
        <a
          className="bi-external-link bi-external-link--mod-topright"
          href={DEMARCHES_SIMPLIFIEES_OFFERER_RIB_UPLOAD_PROCEDURE_URL}
          rel="noopener noreferrer"
          target="_blank"
        >
          <Icon
            alt=""
            svg="ico-external-site"
          />
          {'Modifier'}
        </a>
        <p className="bi-subtitle">
          {
            'Les coordonnées bancaires ci-dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :'
          }
        </p>
        <div className="op-detail">
          <span>
            {'IBAN : '}
          </span>
          <span>
            {offerer.iban}
          </span>
        </div>
        <div className="op-detail">
          <span>
            {'BIC : '}
          </span>
          <span>
            {offerer.bic}
          </span>
        </div>
      </Fragment>
    ) : offerer.demarchesSimplifieesApplicationId ? (
      <div className="bi-banner">
        <p>
          {'Votre dossier est en cours pour cette structure'}
        </p>

        <p>
          <a
            className="bi-external-link"
            href={`https://www.demarches-simplifiees.fr/dossiers/${offerer.demarchesSimplifieesApplicationId}`}
            rel="noopener noreferrer"
            target="_blank"
          >
            <Icon
              alt=""
              svg="ico-external-site"
            />
            {'Accéder au dossier'}
          </a>
        </p>
      </div>
    ) : (
      <Fragment>
        <p className="bi-subtitle">
          {'Aucune coordonnée bancaire renseignée'}
        </p>
        <div className="bi-banner">
          <p>
            {'Renseignez vos coordonnées bancaires pour être remboursé de vos offres éligibles'}
          </p>

          <p>
            <a
              className="bi-external-link"
              href={DEMARCHES_SIMPLIFIEES_OFFERER_RIB_UPLOAD_PROCEDURE_URL}
              rel="noopener noreferrer"
              target="_blank"
            >
              <Icon
                alt=""
                svg="ico-external-site"
              />
              {'Renseignez les coordonnées bancaires de la structure'}
            </a>
          </p>
        </div>
      </Fragment>
    )}
  </div>
)

BankInformation.propTypes = {
  offerer: PropTypes.instanceOf(Offerer).isRequired,
}

export default BankInformation
