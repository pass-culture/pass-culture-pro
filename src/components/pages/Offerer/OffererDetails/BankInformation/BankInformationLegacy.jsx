import React, { Fragment } from 'react'
import { Offerer } from '../Offerer'
import PropTypes from 'prop-types'

const BankInformationLegacy = ({ offerer }) => (
  <Fragment>
    <div className="section">
      <h2 className="main-list-title">
        {'Informations bancaires'}
        <span className="is-pulled-right is-size-7 has-text-grey">
          {!offerer.adminUserOfferer &&
            "Vous avez besoin d'être administrateur de la structure pour modifier ces informations."}
        </span>
      </h2>
      {!offerer.areBankInformationProvided && (
        <p className="bank-instructions-label">
          {
            'Le pass Culture vous contactera prochainement afin d’enregistrer vos coordonnées bancaires. Une fois votre BIC / IBAN renseigné, ces informations apparaitront ci-dessous.'
          }
        </p>
      )}
    </div>
    <div>
      <div>
        <label>
          {'BIC : '}
        </label>
        <span>
          {offerer.bic}
        </span>
      </div>
      <div>
        <label>
          {'IBAN : '}
        </label>
        <span>
          {offerer.iban}
        </span>
      </div>
    </div>
  </Fragment>
)

BankInformationLegacy.propTypes = {
  offerer: PropTypes.instanceOf(Offerer).isRequired,
}

export default BankInformationLegacy
