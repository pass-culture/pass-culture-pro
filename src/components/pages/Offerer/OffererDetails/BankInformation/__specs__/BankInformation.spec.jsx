import React from 'react'
import { shallow } from 'enzyme'
import BankInformation from '../BankInformation'
import { Offerer } from '../../Offerer'

describe('src | components | pages | Offerer | BankInformation ', () => {
  const offerer = new Offerer({
    id: 'AA',
    name: 'fake offerer name',
    address: 'fake address',
    bic: 'ABC',
    iban: 'DEF',
  })

  let props
  beforeEach(() => {
    props = { offerer }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<BankInformation {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  it('should render instruction block information are not provided', () => {
    // Given
    const offererWithoutBI = new Offerer({
      id: 'AA',
      name: 'fake offerer name',
      address: 'fake address',
      bic: null,
      iban: null,
    })

    // when
    const wrapper = shallow(<BankInformation
      {...props}
      offerer={offererWithoutBI}
                            />)

    // then
    const bankInstructions = wrapper.find('.bi-instructions')
    expect(bankInstructions).toHaveLength(1)
    expect(bankInstructions.text()).toBe(
      'Renseigner vos coordonnées bancaires pour être remboursé de vos offres éligibles'
    )
  })

  it('should not render instruction block information are provided', () => {
    // when
    const wrapper = shallow(<BankInformation {...props} />)

    // then
    const bankInstructions = wrapper.find('.bi-instructions')
    expect(bankInstructions).toHaveLength(0)
  })
})
