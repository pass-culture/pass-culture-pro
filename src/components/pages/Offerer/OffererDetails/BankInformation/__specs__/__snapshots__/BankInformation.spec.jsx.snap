// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offerer | BankInformation  should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BankInformation
    offerer={
      Offerer {
        "address": "fake address",
        "adminUserOfferer": undefined,
        "bic": "ABC",
        "city": "",
        "iban": "DEF",
        "id": "AA",
        "name": "fake offerer name",
        "postalCode": "",
        "siren": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="main-list-title"
        >
          Coordonnées bancaires de la structure
        </h2>,
        <React.Fragment>
          <a
            className="bi-external-link bi-external-link--mod-topright"
            href="https://www.demarches-simplifiees.fr/commencer/undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="external-site"
              src="http://localhost/icons/ico-external-site.svg"
            />
            <b>
              Modifier
            </b>
          </a>
          <p
            className="bi-subtitle"
          >
            Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :
          </p>
          <div
            className="op-detail"
          >
            <label>
              IBAN : 
            </label>
            <span>
              DEF
            </span>
          </div>
          <div
            className="op-detail"
          >
            <label>
              BIC : 
            </label>
            <span>
              ABC
            </span>
          </div>
        </React.Fragment>,
      ],
      "className": "section bank-information",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Coordonnées bancaires de la structure",
          "className": "main-list-title",
        },
        "ref": null,
        "rendered": "Coordonnées bancaires de la structure",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <a
              className="bi-external-link bi-external-link--mod-topright"
              href="https://www.demarches-simplifiees.fr/commencer/undefined"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="external-site"
                src="http://localhost/icons/ico-external-site.svg"
              />
              <b>
                Modifier
              </b>
            </a>,
            <p
              className="bi-subtitle"
            >
              Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :
            </p>,
            <div
              className="op-detail"
            >
              <label>
                IBAN : 
              </label>
              <span>
                DEF
              </span>
            </div>,
            <div
              className="op-detail"
            >
              <label>
                BIC : 
              </label>
              <span>
                ABC
              </span>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="external-site"
                  src="http://localhost/icons/ico-external-site.svg"
                />,
                <b>
                  Modifier
                </b>,
              ],
              "className": "bi-external-link bi-external-link--mod-topright",
              "href": "https://www.demarches-simplifiees.fr/commencer/undefined",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "external-site",
                  "src": "http://localhost/icons/ico-external-site.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Modifier",
                },
                "ref": null,
                "rendered": "Modifier",
                "type": "b",
              },
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :",
              "className": "bi-subtitle",
            },
            "ref": null,
            "rendered": "Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  IBAN : 
                </label>,
                <span>
                  DEF
                </span>,
              ],
              "className": "op-detail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "IBAN : ",
                },
                "ref": null,
                "rendered": "IBAN : ",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "DEF",
                },
                "ref": null,
                "rendered": "DEF",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  BIC : 
                </label>,
                <span>
                  ABC
                </span>,
              ],
              "className": "op-detail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "BIC : ",
                },
                "ref": null,
                "rendered": "BIC : ",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ABC",
                },
                "ref": null,
                "rendered": "ABC",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="main-list-title"
          >
            Coordonnées bancaires de la structure
          </h2>,
          <React.Fragment>
            <a
              className="bi-external-link bi-external-link--mod-topright"
              href="https://www.demarches-simplifiees.fr/commencer/undefined"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="external-site"
                src="http://localhost/icons/ico-external-site.svg"
              />
              <b>
                Modifier
              </b>
            </a>
            <p
              className="bi-subtitle"
            >
              Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :
            </p>
            <div
              className="op-detail"
            >
              <label>
                IBAN : 
              </label>
              <span>
                DEF
              </span>
            </div>
            <div
              className="op-detail"
            >
              <label>
                BIC : 
              </label>
              <span>
                ABC
              </span>
            </div>
          </React.Fragment>,
        ],
        "className": "section bank-information",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Coordonnées bancaires de la structure",
            "className": "main-list-title",
          },
          "ref": null,
          "rendered": "Coordonnées bancaires de la structure",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <a
                className="bi-external-link bi-external-link--mod-topright"
                href="https://www.demarches-simplifiees.fr/commencer/undefined"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="external-site"
                  src="http://localhost/icons/ico-external-site.svg"
                />
                <b>
                  Modifier
                </b>
              </a>,
              <p
                className="bi-subtitle"
              >
                Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :
              </p>,
              <div
                className="op-detail"
              >
                <label>
                  IBAN : 
                </label>
                <span>
                  DEF
                </span>
              </div>,
              <div
                className="op-detail"
              >
                <label>
                  BIC : 
                </label>
                <span>
                  ABC
                </span>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="external-site"
                    src="http://localhost/icons/ico-external-site.svg"
                  />,
                  <b>
                    Modifier
                  </b>,
                ],
                "className": "bi-external-link bi-external-link--mod-topright",
                "href": "https://www.demarches-simplifiees.fr/commencer/undefined",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "external-site",
                    "src": "http://localhost/icons/ico-external-site.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Modifier",
                  },
                  "ref": null,
                  "rendered": "Modifier",
                  "type": "b",
                },
              ],
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :",
                "className": "bi-subtitle",
              },
              "ref": null,
              "rendered": "Les coordonnées bancaires ci dessous seront attribuées à tous les lieux sans coordonnées bancaires propres :",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    IBAN : 
                  </label>,
                  <span>
                    DEF
                  </span>,
                ],
                "className": "op-detail",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "IBAN : ",
                  },
                  "ref": null,
                  "rendered": "IBAN : ",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "DEF",
                  },
                  "ref": null,
                  "rendered": "DEF",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    BIC : 
                  </label>,
                  <span>
                    ABC
                  </span>,
                ],
                "className": "op-detail",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "BIC : ",
                  },
                  "ref": null,
                  "rendered": "BIC : ",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ABC",
                  },
                  "ref": null,
                  "rendered": "ABC",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
